### PR TITLE
express-post-json

### PR DESCRIPTION
**Pull request by junoulee**
image

### CODE REVIEW

File: express-post-json/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray = [];
+  for (const key in grades) {
+    gradesArray.push(grades[key]);
+  }
+  res.json(gradesArray);
+});
+
+express.json();
```
Comment (RobertGardner) [inline express-post-json/index.js:15]: These lines should come before `app.get`, `app.post`, etc.

File: express-post-json/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  req.body.id = nextId;
+  grades[nextId] = req.body;
+  nextId++;
+  res.status(201).send(grades);
```
Comment (RobertGardner) [inline express-post-json/index.js:22]: The spec says "The server should respond with the same grade object it received, plus its `id` using the status code `201`." which means just the newly-created grade should be returned, not the entire set of grades.

**Review by RobertGardner — CHANGES_REQUESTED** [review]

File: express-post-json/index.js (hunk 2 above)
Comment (junoulee) [reply express-post-json/index.js]: image

File: express-post-json/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const express = require('express');
+const app = express();
+express.json();
```
Comment (RobertGardner) [inline express-post-json/index.js:3]: This line of code doesn't seem to do anything?

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**